app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb

```diff
@@ -58,19 +56,52 @@ def raw_safe_delete_volume
     EmsRefresh.queue_refresh(ext_management_system)
   end
 
-  def self.params_for_create(provider)
-    services = provider.storage_services.map { |service| {:value => service.ems_ref, :label => service.name} }
+  def params_for_update
+    ems = ExtManagementSystem.find(ems_id)
+    service_value = ems.storage_services.detect { {:value => id, :label => name} }[:name]
```

Comment:
I don't think this is doing anything `.detect { {:value => id, :label => name} }`  if anything it is likely just returning the first service always because `{:value => id, :label => name}` is truthy

Not sure what you're trying to do, but is it `ems.storage_services.find_by(:value => id, :label => name)` ?

Reply:
It was suppose to be:
`ems = ExtManagementSystem.find(ems_id)`
`storage_service_id = ems.cloud_volumes.find_by(:id => id)[:storage_service_id]`
`service_value = ems.storage_services.find_by(:id => storage_service_id)[:name]`

I pushed the correct code now